File: TextosJapones/src/main/java/org/jisho/textosJapones/model/entities/mangaextractor/MangaCapitulo.java
package org.jisho.textosJapones.model.entities.mangaextractor;

import com.google.gson.annotations.Expose;
import org.jisho.textosJapones.model.entities.Manga;
import org.jisho.textosJapones.model.entities.VocabularioExterno;
import org.jisho.textosJapones.model.enums.Language;

import java.util.*;

public class MangaCapitulo extends Manga {

	private UUID id;
	@Expose
	private String manga;
	@Expose
	private Integer volume;
	@Expose
	private Float capitulo;
	@Expose
	private Language lingua;
	@Expose
	private String scan;
	@Expose
	private List<MangaPagina> paginas;
	@Expose
	private Boolean extra;
	@Expose
	private Boolean raw;
	@Expose
	private Set<VocabularioExterno> vocabularios;

	public UUID getId() {
		return id;
	}

	public void setId(UUID id) {
		this.id = id;
	}

	public String getManga() {
		return manga;
	}

	public void setManga(String manga) {
		this.manga = manga;
	}

	public Integer getVolume() {
		return volume;
	}

	public void setVolume(Integer volume) {
		this.volume = volume;
	}

	public Float getCapitulo() {
		return capitulo;
	}

	public void setCapitulo(Float capitulo) {
		this.capitulo = capitulo;
	}

	public Language getLingua() {
		return lingua;
	}

	public void setLingua(Language lingua) {
		this.lingua = lingua;
	}

	public String getScan() {
		return scan;
	}

	public void setScan(String scan) {
		this.scan = scan;
	}

	public Boolean isExtra() {
		return extra;
	}

	public void setExtra(Boolean extra) {
		this.extra = extra;
	}

	public Boolean isRaw() {
		return raw;
	}

	public void setRaw(Boolean raw) {
		this.raw = raw;
	}

	public List<MangaPagina> getPaginas() {
		return paginas;
	}

	public void setPaginas(List<MangaPagina> paginas) {
		this.paginas = paginas;
	}

	public void addPaginas(MangaPagina pagina) {
		this.paginas.add(pagina);
	}

	public Set<VocabularioExterno> getVocabularios() {
		return vocabularios;
	}

	public void setVocabularios(Set<VocabularioExterno> vocabularios) {
		this.vocabularios = vocabularios;
	}

	public void addVocabulario(VocabularioExterno vocabulario) {
		this.vocabularios.add(vocabulario);
	}

	public MangaCapitulo() {
		super();
		this.id = null;
		this.manga = "";
		this.volume = 0;
		this.capitulo = 0F;
		this.lingua = Language.PORTUGUESE;
		this.scan = "";
		this.vocabularios = new HashSet<>();
		this.processar = true;
		this.paginas = new ArrayList<>();
	}

	public MangaCapitulo(UUID id, String manga, Integer volume, Float capitulo, Language lingua, String scan, Boolean extra, Boolean raw) {
		super(manga, volume, capitulo);
		this.id = id;
		this.manga = manga;
		this.volume = volume;
		this.capitulo = capitulo;
		this.lingua = lingua;
		this.scan = scan;
		this.vocabularios = new HashSet<>();
		this.processar = true;
		this.paginas = new ArrayList<>();
		this.extra = extra;
		this.raw = raw;
	}

	public MangaCapitulo(UUID id, String manga, Integer volume, Float capitulo, Language lingua, String scan,
			Boolean extra, Boolean raw, List<MangaPagina> paginas) {
		super(manga, volume, capitulo);
		this.id = id;
		this.manga = manga;
		this.volume = volume;
		this.capitulo = capitulo;
		this.lingua = lingua;
		this.scan = scan;
		this.vocabularios = new HashSet<>();
		this.processar = true;
		this.paginas = paginas;
		this.extra = extra;
		this.raw = raw;
	}

	public MangaCapitulo(UUID id, String manga, Integer volume, Float capitulo, Language lingua, String scan,
			Set<VocabularioExterno> vocabularios) {
		super(manga, volume, capitulo);
		this.id = id;
		this.manga = manga;
		this.volume = volume;
		this.capitulo = capitulo;
		this.lingua = lingua;
		this.scan = scan;
		this.vocabularios = vocabularios;
		this.processar = true;
		this.paginas = new ArrayList<MangaPagina>();
	}

	public MangaCapitulo(UUID id, String manga, Integer volume, Float capitulo, Language lingua, String scan,
			Boolean extra, Boolean raw, Set<VocabularioExterno> vocabularios, List<MangaPagina> paginas) {
		super(manga, volume, capitulo);
		this.id = id;
		this.manga = manga;
		this.volume = volume;
		this.capitulo = capitulo;
		this.lingua = lingua;
		this.scan = scan;
		this.paginas = paginas;
		this.vocabularios = vocabularios;
		this.processar = true;
		this.extra = extra;
		this.raw = raw;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		if (!super.equals(o)) return false;
		MangaCapitulo that = (MangaCapitulo) o;
		return Objects.equals(id, that.id) && Objects.equals(manga, that.manga) && Objects.equals(volume, that.volume) && Objects.equals(capitulo, that.capitulo) && lingua == that.lingua && Objects.equals(scan, that.scan) && Objects.equals(extra, that.extra) && Objects.equals(raw, that.raw);
	}

	@Override
	public int hashCode() {
		return Objects.hash(super.hashCode(), id, manga, volume, capitulo, lingua, scan, extra, raw);
	}

	@Override
	public String toString() {
		return "MangaCapitulo [id=" + id + ", capitulo=" + capitulo + ", lingua=" + lingua + ", scan=" + scan
				+ ", paginas=" + paginas + ", extra=" + extra + ", raw=" + raw + "]";
	}

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/model/entities/mangaextractor/MangaCapitulo.kt
package br.com.fenix.processatexto.model.entities.mangaextractor

import br.com.fenix.processatexto.model.entities.Entity
import br.com.fenix.processatexto.model.entities.Manga
import br.com.fenix.processatexto.model.entities.processatexto.VocabularioExterno
import br.com.fenix.processatexto.model.enums.Language
import com.google.gson.annotations.Expose
import java.util.*


data class MangaCapitulo(
    private var id: UUID? = null,
    @Expose override var manga: String = "",
    @Expose override var volume: Int = 0,
    @Expose override var capitulo: Float = 0f,
    @Expose var lingua: Language = Language.PORTUGUESE,
    @Expose var scan: String = "",
    @Expose var paginas: MutableList<MangaPagina> = mutableListOf(),
    @Expose var isExtra: Boolean = false,
    @Expose var isRaw: Boolean = false,
    @Expose var vocabularios: MutableSet<VocabularioExterno> = mutableSetOf()
) : Manga(), Entity<UUID?, MangaCapitulo> {

    override fun getId(): UUID? = id

    fun setId(id: UUID?) {
        this.id = id
    }

    override fun create(id: UUID?): MangaCapitulo {
        TODO("Not yet implemented")
    }

    fun addPaginas(pagina: MangaPagina) = paginas.add(pagina)

    fun addVocabulario(vocabulario: VocabularioExterno) = vocabularios.add(vocabulario)

    /*constructor() : super() {
        this.processar = true
    }*/

    constructor(id: UUID?, manga: String, volume: Int, capitulo: Float, lingua: Language, scan: String, extra: Boolean, raw: Boolean) :
            this() {
        setInitial(manga, volume, capitulo)
        this.id = id
        this.lingua = lingua
        this.scan = scan
        super.isProcessar = true
        isExtra = extra
        isRaw = raw
    }

    constructor(
        id: UUID?, manga: String, volume: Int, capitulo: Float, lingua: Language, scan: String,
        extra: Boolean, raw: Boolean, paginas: MutableList<MangaPagina>
    ) : this() {
        setInitial(manga, volume, capitulo)
        this.id = id
        this.lingua = lingua
        this.scan = scan
        super.isProcessar = true
        this.paginas = paginas
        isExtra = extra
        isRaw = raw
    }

    constructor(
        id: UUID?, manga: String, volume: Int, capitulo: Float, lingua: Language, scan: String,
        vocabularios: MutableSet<VocabularioExterno>
    ) : this() {
        setInitial(manga, volume, capitulo)
        this.id = id
        this.lingua = lingua
        this.scan = scan
        this.vocabularios = vocabularios
        super.isProcessar = true
        paginas = ArrayList<MangaPagina>()
    }

    constructor(
        id: UUID?, manga: String, volume: Int, capitulo: Float, lingua: Language, scan: String,
        extra: Boolean, raw: Boolean, vocabularios: MutableSet<VocabularioExterno>, paginas: MutableList<MangaPagina>
    ) : this() {
        setInitial(manga, volume, capitulo)
        this.id = id
        this.lingua = lingua
        this.scan = scan
        this.paginas = paginas
        this.vocabularios = vocabularios
        super.isProcessar = true
        isExtra = extra
        isRaw = raw
    }

    override fun toString(): String {
        return ("MangaCapitulo [id=" + id + ", capitulo=" + capitulo + ", lingua=" + lingua + ", scan=" + scan
                + ", paginas=" + paginas + ", extra=" + isExtra + ", raw=" + isRaw + "]")
    }

    override fun hashCode(): Int {
        return Objects.hash(super.hashCode(), id, manga, volume, capitulo, lingua, scan, isExtra, isRaw)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MangaCapitulo

        if (id != other.id) return false
        if (manga != other.manga) return false
        if (volume != other.volume) return false
        if (capitulo != other.capitulo) return false
        if (lingua != other.lingua) return false
        if (scan != other.scan) return false
        if (isExtra != other.isExtra) return false
        if (isRaw != other.isRaw) return false

        return true
    }
}